### PR TITLE
GH-394 Tests no longer depend on the legacy provider.

### PR DESCRIPTION
**Pull request by h-vn**
`Net::SSLeay::PEM_get_string_PrivateKey()` uses DES-CBC as its default encryption algorithm. Providers introduced in OpenSSL 3.0 implement DES-CBC in the legacy provider which may not be available on all systems. This commit changes `33_x509_create_cert.t `PEM_get_string_PrivateKey() default encryption testing to skip the test on systems that support providers but do not have the default provider available.

This change does not change the fact that `PEM_get_string_PrivateKey(`) remains broken on systems without the default provider when the default encryption algorithm is used.

When obsolete encryption algorithms are updated, for example by changing DES-CBC to AES-128-CBC, the special handling can be removed and the test can be re-enabled on all systems.

As a reminder for this, test `fail()` is called when `Net::SSLeay::VERSION` reaches 2.x.

Closes GH-394. Relates to GH-402.

### CODE REVIEW

File: t/local/33_x509_create_cert.t [hunk 1]
```diff
@@ -114,7 +107,22 @@ is(Net::SSLeay::X509_NAME_cmp($ca_issuer, $ca_subject), 0, "X509_NAME_cmp");
...
+      # Upcoming Net::SSLeay version 2.00 is likely to remove obsolete
+      # functionality. This includes updating PEM_get_stringPrivateKey
+      # default algorithm from EVP_des_cbc() to, for example,
+      # EVP_aes_128_cbc(). When this is done, this SKIP block and
+      # everything in it, besides the test itself, can be removed.
+      fail("Legacy provider still used with Net::SSLeay version $Net::SSLeay::VERSION") if $Net::SSLeay::VERSION =~ m/^2/s;
+      if (defined &Net::SSLeay::OSSL_PROVIDER_load &&
+	  !Net::SSLeay::OSSL_PROVIDER_load(undef, 'legacy'))
+      {
+	  my $des_warning = 'Warning: No legacy provider. DES in CBC mode used by PEM_get_string_PrivateKey by default is not available';
```
Comment (chrisnovakovic) [inline t/local/33_x509_create_cert.t:120]: Since Test::More prints this to the console, it might be better to shorten it to `No legacy provider` and mention the reason the legacy provider is needed for this test in the comment above.

File: lib/Net/SSLeay.pod [hunk 2]
```diff
@@ -1836,7 +1836,7 @@ Converts/exports X509 certificate to string (PEM format).
 
 =item * PEM_get_string_PrivateKey
 
-B<COMPATIBILITY:> not available in Net-SSLeay-1.45 and before
+B<COMPATIBILITY:> not available in Net-SSLeay-1.45 and before. Requires legacy provider with OpenSSL 3.0 and later as long as DES-CBC is the default encryption algorithm.
```
Comment (chrisnovakovic) [inline lib/Net/SSLeay.pod:1839]:
```suggestion
B<COMPATIBILITY:> not available in Net-SSLeay-1.45 and before. Requires legacy provider with OpenSSL 3.0 and later if the default value of C<$enc_alg> (C<DES_CBC>) is used.
```

I'd suggest rewording this to make it more user-centric - something like:

> Requires legacy provider with OpenSSL 3.0 and later if the default value of `$enc_alg` (`DES_CBC`) is used.

otherwise it might sound like a discussion of an internal implementation detail.

File: t/local/33_x509_create_cert.t (hunk 1 above)
Comment (h-vn) [reply t/local/33_x509_create_cert.t:120]: I can add more comments and shorten and update the `diag()` message to tell to check the comments in the text. Because it currently fails the test, the packager or end user is aware that something does not right when there's no default provider. My idea with a prominent message on the console is to try to ensure that the same person now still notices that something may not work even though the tests are passing.

The failure is sort-of expected behaviour but I want also to make sure that this is not hidden and later causing surprising failures.

Comment (h-vn) [reply t/local/33_x509_create_cert.t:120]: `diag()` is now shorter and tells what's it's for:
```
t/local/32_x509_get_cert_info.t ............. ok       
t/local/33_x509_create_cert.t ............... 1/141 # No legacy provider for PEM_get_string_PrivateKey
t/local/33_x509_create_cert.t ............... ok       
```

**Review by chrisnovakovic — APPROVED** [review]
Looks good, thanks!
